### PR TITLE
new method to GetObjectsWithAttributes

### PR DESCRIPTION
**Pull request by CBaptiste**

### CODE REVIEW

File: Algolia.Search/Index.cs [hunk 1]
```diff
@@ -172,6 +172,37 @@ public Task<JObject> GetObjectsAsync(IEnumerable<String> objectIDs, Cancellation
...
+        /// <returns></returns> 
+        public Task<JObject> GetObjectsWithAttributesAsync(IEnumerable<String> objectIDs, IEnumerable<string> attributesToRetrieve, CancellationToken token = default(CancellationToken))
+        {
+            JArray requests = new JArray();
+            var attributes = "";
+            foreach (string attr in attributesToRetrieve)
+            {
+                if (attributes.Length > 0) {
+                    attributes += ",";
+                    attributes += Uri.EscapeDataString(attr);
```
Comment (Nagriar) [inline Algolia.Search/Index.cs]: Why do you encode the attribute names? We pass it in the body so we shouldn't need to encode it.

Comment (CBaptiste) [inline Algolia.Search/Index.cs]: true !

File: Algolia.Search/Index.cs [hunk 2]
```diff
@@ -172,6 +172,37 @@ public Task<JObject> GetObjectsAsync(IEnumerable<String> objectIDs, Cancellation
         }
 
         /// <summary>
+        /// Get several objects from this index.
+        /// </summary>
+        /// <param name="objectIDs">An array of unique identifiers of the objects to retrieve.</param>
+        /// <returns></returns> 
+        public Task<JObject> GetObjectsWithAttributesAsync(IEnumerable<String> objectIDs, IEnumerable<string> attributesToRetrieve, CancellationToken token = default(CancellationToken))
```
Comment (Nagriar) [inline Algolia.Search/Index.cs]: We could also overload the GetObjects method no?

Comment (CBaptiste) [inline Algolia.Search/Index.cs]: Yes, but it will not be retro-compatible for those that have CancellationToken token != default

Comment (CBaptiste) [inline Algolia.Search/Index.cs]: by convention, it would be the cancelation token as last param, not second

Comment (Nagriar) [inline Algolia.Search/Index.cs]: You can have two prototypes for the same method name in csharp no?

`public Task<JObject> GetObjects(IEnumerable<String> objectIDs, IEnumerable<string> attributesToRetrieve, CancellationToken token = default(CancellationToken))`
and
`public Task<JObject> GetObjects(IEnumerable<String> objectIDs, IEnumerable<string> attributesToRetrieve, CancellationToken token = default(CancellationToken))`

Comment (CBaptiste) [inline Algolia.Search/Index.cs]: you mean
`public Task<JObject> GetObjects(IEnumerable<String> objectIDs, IEnumerable<string> attributesToRetrieve, CancellationToken token = default(CancellationToken))``and 
`public Task<JObject> GetObjects(IEnumerable<String> objectIDs, CancellationToken token = default(CancellationToken))` ? yes you can, it was to make it more understadable but ok :)

Comment (Nagriar) [inline Algolia.Search/Index.cs]: I don't know the C# convention so I'll let you decide what is the best name for it :)

File: Algolia.Search.Test/AlgoliaClientTest.cs [hunk 3]
```diff
@@ -47,7 +47,7 @@ public void clearTest()
         public void TestInitialize()
         {
             _testApiKey = Environment.GetEnvironmentVariable("ALGOLIA_API_KEY");
-            _testApplicationID = Environment.GetEnvironmentVariable("ALGOLIA_APPLICATION_ID");
+            _testApplicationID = Environment.GetEnvironmentVariable("ALGOLIA_API_KEY");
             _client = new AlgoliaClient(_testApplicationID, _testApiKey);
             _index = _client.InitIndex(safe_name("àlgol?à-csharp"));
         }
```
Comment (CBaptiste) [inline Algolia.Search.Test/AlgoliaClientTest.cs]: oops

**Pull request merged**